src/utils/rfq_blocked_address_utils.ts

```diff
@@ -0,0 +1,64 @@
+import { Connection } from 'typeorm/connection/Connection';
+
+import { BlockedAddressEntity } from '../entities/BlockedAddressEntity';
+
+/**
+ * RfqBlockedAddressUtils helps manage the RFQ blocked addresses
+ */
+export class RfqBlockedAddressUtils {
+    private _blocked: Set<string>;
+    private _expiresAt: number;
+    private _updatePromise: Promise<void> | undefined;
+    private _updating: boolean;
+    private readonly _ttlMs: number;
+
+    constructor(private readonly _connection: Connection, initialBlacklist: Set<string>, ttlMs: number) {
+        this._blocked = initialBlacklist;
+        this._ttlMs = ttlMs;
+        this._updating = false;
+        this._expiresAt = Date.now().valueOf(); // cache expires immediately
+    }
+
+    /**
+     * isBlocked returns whether an address is blocked from the cache
+     * NOTE: In the background, it also updates the blocked set if the cache is expired
+     */
+    public isBlocked(address: string): boolean {
```

Comment:
I imagine we'll just use the isBlocked function, wdyt?

Reply:
correct - the `isBlockedAsync` version is just there to make the test deterministic

Reply:
removed it - got something a little better